internal/chain/reward.go

```diff
@@ -27,22 +28,53 @@ func ballotResultBeaconchain(
 	return availability.BallotResult(bc, header, shard.BeaconChainShardID)
 }
 
+var (
+	votingPowerCache singleflight.Group
+)
+
+const (
+	votepowerCacheTimeLimit = 3 * time.Minute
+)
+
+func lookupVotingPower(
+	epoch *big.Int, subComm *shard.Committee,
+) (*votepower.Roster, error) {
+	key := fmt.Sprintf("%s-%d", epoch.String(), subComm.ShardID)
+	results, err, _ := votingPowerCache.Do(
+		key, func() (interface{}, error) {
+			votingPower, err := votepower.Compute(subComm)
+			if err != nil {
+				return nil, err
+			}
+			go func() {
+				time.Sleep(votepowerCacheTimeLimit)
```

Comment:
why forget every 3 minutes? We need the voting power for the whole epoch right?

Reply:
Just picking some period till you help me pick a better value